simtools/corsika/corsika_default_config.py

```diff
@@ -0,0 +1,217 @@
+from collections import defaultdict
+
+import astropy.units as u
+from scipy.interpolate import interp1d
+
+
+class CorsikaDefaultConfig:
+    """
+    This class contains the default configuration for CORSIKA parameters for
+    the various primary particles. It includes all basic dependencies on zenith angles, etc.
+    """
+
+    def __init__(self, primary=None, zenith_angle=None):
+        """
+        Initialize CorsikaDefaultConfig.
+        """
+        self._primary = primary
+        self._zenith_angle = zenith_angle
+        self._energy_slope = -2.0
+        self.energy_ranges = self._define_hardcoded_energy_ranges()
+        self.number_of_showers = self._define_hardcoded_number_of_showers()
+
+    @property
+    def primary(self):
+        """
+        Primary particle.
+        """
+        return self._primary
+
+    @primary.setter
+    def primary(self, primary):
+        """
+        Set primary particle.
+        """
+        supported_primaries = [
+            "gamma",
+            "gamma_diffuse",
+            "electron",
+            "proton",
+            "muon",
+            "helium",
+            "nitrogen",
+            "silicon",
+            "iron",
+        ]
+        if primary not in supported_primaries:
+            raise ValueError(
+                f"Invalid primary particle: {primary}. Must be one of {supported_primaries}"
+            )
+        self._primary = primary
+
+    @property
+    def zenith_angle(self):
+        """
+        Zenith angle.
+        """
+        return self._zenith_angle
+
+    @zenith_angle.setter
+    @u.quantity_input(zenith_angle=u.deg)
+    def zenith_angle(self, zenith_angle):
+        """
+        Set zenith angle.
+        """
+        allowed_zenith_angle_interval = [20.0, 60.0] * u.deg
```

Comment:
Why is the default allowed zenith angle interval 20-60, and not 0-60 (or even >60, as the required elevation range of CTA is larger)? Is it because energy range, etc are only hardwired for these values?

Reply:
Yes, I do not have optimised values of energy ranges and number of events for anything below 20 degrees or above 60 degrees. Below 20 degrees I could perhaps "guess" because it is not much different to 20 degrees, but I don't see a need for it right now. Above 60 degrees would require an actual (short) study.